### PR TITLE
Provide a warning for Layer Keys

### PR DESCRIPTION
**Pull request by yanfali**
- enhance space key to support optional title and icon and hover
 - provide a warning for the layer keys that using them will overwrite
   the key above to KC_TRNS to prevent layer lock
 - add fa-exclamation-triangle icon

@fauxpark @noroadsleft @mechmerlin

### CODE REVIEW

**Comment by yanfali** [comment]
Warning Icon

Warning Text

**Comment by yanfali** [comment]
Issue #282

**Comment by noroadsleft** [comment]
I'm thinking saying "on the target layer" instead of "above" is better. "Above" makes it sound like the KC_TRNS is forced onto the next layer, but the next layer might not be the target layer (user could be doing MO(3) from Layer 0).

Maybe in a future iteration, we could notify the user which keycode was removed, if the forced KC_TRNS overwrites a key.

File: src/store/modules/keycodes/quantum.js [hunk 1]
```diff
@@ -31,7 +31,14 @@ export default [
...
   },
 
-  { label: 'Layer and Layer Tap functions', width: 'label' },
+  {
+    label: 'Layer and Layer Tap functions',
+    icon: 'exclamation-triangle',
+    iconClass: 'warning',
+    width: 'label',
+    title:
+      'Note: Layer keys override the position on the target layer to TRNS to avoid layer lock'
```
Comment (fauxpark) [inline src/store/modules/keycodes/quantum.js]:
```suggestion
      'Note: Layer keys overwrite the key at the same position on the target layer to KC_TRNS to avoid layer lock'
```

"Override" implies the key will return to its original assignment if the layer key is removed.

Comment (yanfali) [reply src/store/modules/keycodes/quantum.js]: Sorry that makes no sense unless we allow undo.

Comment (fauxpark) [reply src/store/modules/keycodes/quantum.js]: Uh, it *does* make sense *because* undo is not supported (yet).

Comment (yanfali) [reply src/store/modules/keycodes/quantum.js]: I do like the overwrite better though. I will need to test to see if it's too verbose for the screen. It's quite long already and on a contricted width it may be too wordy

Comment (yanfali) [reply src/store/modules/keycodes/quantum.js]: Agree to disagree. But thank you for the input.

Comment (fauxpark) [reply src/store/modules/keycodes/quantum.js]: I'm confused as to what we're disagreeing on... you say you like "overwrite" better, which is exactly what the suggestion does.

Comment (fauxpark) [reply src/store/modules/keycodes/quantum.js]: For clarity, my comment below the suggestion is about the current behaviour, where if you remove a MO(), the transparent key above it is not also removed or reverted back to what it was before the placement of the MO() turned it into KC_TRNS. I call that "overwriting", not "overriding".

Comment (yanfali) [reply src/store/modules/keycodes/quantum.js]: Thanks very much.
